### PR TITLE
Cleaner logging

### PR DESCRIPTION
**Pull request by ylt**
Addresses https://github.com/openactive/openactive-test-suite/issues/46

- Cleans out the titles so that the full test path isn't references
- Removes out sections without any info to display
- Renders out errors

Couple of other changes:
- Updated project dependencies to non-vulnerable versions
- ANSI stripping

### CODE REVIEW

**Comment by nickevansuk** [comment]
Hmmm.... there still doesn't seem to be any information about why C1 failed here:



# 

A successful end to end booking.

## Test setup
### random-opportunity Request
GET http://localhost:3000/get-random-opportunity?type=ScheduledSession

---

Response status code: 200 OK. Responded in 46.4425ms
```json
{
    "@context": "https://openactive.io/",
    "@type": "ScheduledSession",
    "@id": "https://localhost:44365/api/identifiers/scheduled-sessions/0861CAB18000417/events/2020-03-16T18%3A00%3A00Z/860267824"
}
```
### get-match Request
GET http://localhost:3000/get-cached-opportunity/https%3A%2F%2Flocalhost%3A44365%2Fapi%2Fidentifiers%2Fscheduled-sessions%2F0861CAB18000417%2Fevents%2F2020-03-16T18%253A00%253A00Z%2F860267824

---

Response status code: 200 OK. Responded in 12.0238ms
```json
{
    "data": {
        "@context": [
            "https://openactive.io/",
            "https://openactive.io/ns-beta"
        ],
        "@type": "ScheduledSession",
        "@id": "https://localhost:44365/api/identifiers/scheduled-sessions/0861CAB18000417/events/2020-03-16T18%3A00%3A00Z/860267824",
        "identifier": 860267824,
        "superEvent": {
            "@context": "https://openactive.io/",
            "@type": "SessionSeries",
            "@id": "https://localhost:44365/api/identifiers/session-series/0861CAB18000417",
            "identifier": "0861CAB18000417",
            "name": "Pure Abs 18:00 Mon",
            "category": [
                "Group Exercise 16+ Yrs"
            ],
            "duration": "PT45M",
            "eventSchedule": [
                {
                    "@type": "PartialSchedule",
                    "byDay": [
                        "https://schema.org/Monday"
                    ],
                    "duration": "PT45M",
                    "endTime": "18:45",
                    "startDate": "2018-01-08",
                    "endDate": "2022-12-31",
                    "startTime": "18:00"
                }
            ],
            "location": {
                "@type": "Place",
                "identifier": "0086",
                "name": "Saltburn Leisure Centre",
                "address": {
                    "@type": "PostalAddress"
                }
            },
            "offers": [
                {
                    "@type": "Offer",
                    "@id": "https://localhost:44365/api/identifiers/session-series/0861CAB18000417#/offers/OX-AD",
                    "identifier": "OX-AD",
                    "name": "Oxygen - Adult",
                    "description": "Grp Ex - Pure Abs 45",
                    "availableChannel": [
                        "https://openactive.io/OpenBookingPrepayment"
                    ],
                    "price": 4.6,
                    "priceCurrency": "GBP",
                    "validFromBeforeStartDate": "P5D"
                },
                {
                    "@type": "Offer",
                    "@id": "https://localhost:44365/api/identifiers/session-series/0861CAB18000417#/offers/OX-SNR",
                    "identifier": "OX-SNR",
                    "name": "Oxygen - Senior",
                    "description": "Grp Ex - Pure Abs 45",
                    "ageRange": {
                        "@type": "QuantitativeValue",
                        "minValue": 50
                    },
                    "price": 2.4,
                    "priceCurrency": "GBP",
                    "validFromBeforeStartDate": "P5D"
                }
            ],
            "organizer": {
                "@type": "Organization",
                "@id": "https://localhost:44365/api/identifiers/seller",
                "taxMode": "https://openactive.io/TaxGross"
            },
            "superEvent": {
                "@type": "EventSeries",
                "identifier": 49,
                "name": "Pure Abs",
                "description": "A short fun session aimed at honing and toning your stomach muscles.",
                "activity": [
                    {
                        "@type": "Concept",
                        "@id": "https://openactive.io/activity-list#741097ba-21a0-4b7b-8099-86ddb562458a",
                        "inScheme": "https://openactive.io/activity-list",
                        "prefLabel": "abscore"
                    }
                ],
                "ageRange": {
                    "@type": "QuantitativeValue",
                    "maxValue": 100,
                    "minValue": 16
                },
                "isCoached": true,
                "organizer": {
                    "@type": "Organization",
                    "name": "EveryoneActive",
                    "legalName": "EveryoneActive"
                }
            }
        },
        "duration": "PT45M",
        "maximumAttendeeCapacity": 25,
        "remainingAttendeeCapacity": 25,
        "startDate": "2020-03-16T18:00:00+00:00",
        "endDate": "2020-03-16T18:45:00+00:00",
        "beta:sportsActivityLocation": [
            {
                "@type": "SportsActivityLocation",
                "name": "Studio 1"
            }
        ]
    }
}
```
## C1
### C1 Request
PUT https://localhost:44365/api/openbooking/order-quote-templates/2fb72701-685f-5ecc-ac5a-206a788e960a
```json
{
    "@context": "https://openactive.io/",
    "@type": "OrderQuote",
    "brokerRole": "https://openactive.io/AgentBroker",
    "broker": {
        "@type": "Organization",
        "name": "MyFitnessApp",
        "url": "https://myfitnessapp.example.com",
        "description": "A fitness app for all the community",
        "logo": {
            "@type": "ImageObject",
            "url": "http://data.myfitnessapp.org.uk/images/logo.png"
        },
        "address": {
            "@type": "PostalAddress",
            "streetAddress": "Alan Peacock Way",
            "addressLocality": "Village East",
            "addressRegion": "Middlesbrough",
            "postalCode": "TS4 3AE",
            "addressCountry": "GB"
        }
    },
    "seller": {
        "@type": "Organization",
        "@id": "https://localhost:44365/api/identifiers/seller"
    },
    "orderedItem": [
        {
            "@type": "OrderItem",
            "position": 0,
            "acceptedOffer": {
                "@type": "Offer",
                "@id": "https://localhost:44365/api/identifiers/session-series/0861CAB18000417#/offers/OX-AD"
            },
            "orderedItem": {
                "@type": "ScheduledSession",
                "@id": "https://localhost:44365/api/identifiers/scheduled-sessions/0861CAB18000417/events/2020-03-16T18%3A00%3A00Z/860267824"
            }
        }
    ]
}
```
### Specs
 * ❌️ should return 200 on success
    - Error: Expected C1 request to be successful
 * ❌️ should return the type 'ScheduledSession'
    - Error: Expected C1 request to be successful
 * ❌️ C1 Order or OrderQuote should have one orderedItem
    - Error: Expected C1 request to be successful
## C1 >> validation of C1
### Specs
 * ❌️ passes validation checks
    - Error: FAILURE: $: Only objects are supported for validation. Please only submit single objects for validation: either an object conforming to the [Modelling Specification](https://www.openactive.io/modelling-opportunity-data/), or an [RPDE](https://www.openactive.io/realtime-paged-data-exchange/) feed root object.
### Validations
 * ❌️ Only objects are supported for validation. Please only submit single objects for validation: either an object conforming to the [Modelling Specification](https://www.openactive.io/modelling-opportunity-data/), or an [RPDE](https://www.openactive.io/realtime-paged-data-exchange/) feed root object.
## C2
### Specs
 * ❌️ should return 200 on success
    - Error: Pre-requisite step failed: C1 failed
    - Error: Expected C2 request to be successful
 * ❌️ Order or OrderQuote should have one orderedItem
    - Error: Pre-requisite step failed: C1 failed
    - Error: Expected C2 request to be successful
## C2 >> validation of C2
### Specs
 * ❌️ passes validation checks
    - Error: Pre-requisite step failed: C1 failed
    - Error: No response to validate
## B
### Specs
 * ❌️ should return 200 on success
    - Error: Pre-requisite step failed: C1 failed
    - Error: Expected B request to be successful.
 * ❌️ B Order or OrderQuote should have one orderedItem
    - Error: Pre-requisite step failed: C1 failed
    - Error: Expected B request to be successful.
 * ❌️ Result from B should OrderItemConfirmed orderItemStatus
    - Error: Pre-requisite step failed: C1 failed
    - Error: Expected B request to be successful.
## B >> validation of B
### Specs
 * ❌️ passes validation checks
    - Error: Pre-requisite step failed: C1 failed
    - Error: No response to validate

**Comment by nickevansuk** [comment]
Also notice that in the output above the timeout errors (https://github.com/openactive/openactive-test-suite/issues/46#issuecomment-597538590) aren’t showing up for me

**Comment by ylt** [comment]
Oh, that's really strange - the command output shouldn't be much different

**Comment by nickevansuk** [comment]
Also I'm still getting `This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.`, even when the prereqs all fail so there should be nothing outstanding

Command:
`npm test -- test/flows/book-only/book-only-success/book-random-test.js`

Output:
```
A successful end to end booking.

See `./output/Book Only Book Only Success for random ScheduledSession.md` for more detailed info.

>> Book Only >> Book Only Success for random ScheduledSession
>> Book Only >> Book Only Success for random ScheduledSession >> C1
  ❌️ should return 200 on success
      Error: Pre-requisite step failed: event not found
      Error: Expected C1 request to be successful
  ❌️ should return the type 'ScheduledSession'
      Error: Pre-requisite step failed: event not found
      Error: Expected C1 request to be successful
  ❌️ C1 Order or OrderQuote should have one orderedItem
      Error: Pre-requisite step failed: event not found
      Error: Expected C1 request to be successful
>> Book Only >> Book Only Success for random ScheduledSession >> C1 >> validation of C1
  ❌️ passes validation checks
      Error: Pre-requisite step failed: event not found
      Error: No response to validate
>> Book Only >> Book Only Success for random ScheduledSession >> C2
  ❌️ should return 200 on success
      Error: Pre-requisite step failed: event not found
      Error: Expected C2 request to be successful
  ❌️ Order or OrderQuote should have one orderedItem
      Error: Pre-requisite step failed: event not found
      Error: Expected C2 request to be successful
>> Book Only >> Book Only Success for random ScheduledSession >> C2 >> validation of C2
  ❌️ passes validation checks
      Error: Pre-requisite step failed: event not found
      Error: No response to validate
>> Book Only >> Book Only Success for random ScheduledSession >> B
  ❌️ should return 200 on success
      Error: Pre-requisite step failed: event not found
      Error: Expected B request to be successful.
  ❌️ B Order or OrderQuote should have one orderedItem
      Error: Pre-requisite step failed: event not found
      Error: Expected B request to be successful.
  ❌️ Result from B should OrderItemConfirmed orderItemStatus
      Error: Pre-requisite step failed: event not found
      Error: Expected B request to be successful.
>> Book Only >> Book Only Success for random ScheduledSession >> B >> validation of B
  ❌️ passes validation checks
      Error: Pre-requisite step failed: event not found
      Error: No response to validate

======



A successful end to end booking.

See `./output/Book Only Book Only Success for random Slot.md` for more detailed info.

>> Book Only >> Book Only Success for random Slot
>> Book Only >> Book Only Success for random Slot >> C1
  ❌️ should return 200 on success
      Error: Pre-requisite step failed: event not found
      Error: Expected C1 request to be successful
  ❌️ should return the type 'Slot'
      Error: Pre-requisite step failed: event not found
      Error: Expected C1 request to be successful
  ❌️ C1 Order or OrderQuote should have one orderedItem
      Error: Pre-requisite step failed: event not found
      Error: Expected C1 request to be successful
>> Book Only >> Book Only Success for random Slot >> C1 >> validation of C1
  ❌️ passes validation checks
      Error: Pre-requisite step failed: event not found
      Error: No response to validate
>> Book Only >> Book Only Success for random Slot >> C2
  ❌️ should return 200 on success
      Error: Pre-requisite step failed: event not found
      Error: Expected C2 request to be successful
  ❌️ Order or OrderQuote should have one orderedItem
      Error: Pre-requisite step failed: event not found
      Error: Expected C2 request to be successful
>> Book Only >> Book Only Success for random Slot >> C2 >> validation of C2
  ❌️ passes validation checks
      Error: Pre-requisite step failed: event not found
      Error: No response to validate
>> Book Only >> Book Only Success for random Slot >> B
  ❌️ should return 200 on success
      Error: Pre-requisite step failed: event not found
      Error: Expected B request to be successful.
  ❌️ B Order or OrderQuote should have one orderedItem
      Error: Pre-requisite step failed: event not found
      Error: Expected B request to be successful.
  ❌️ Result from B should OrderItemConfirmed orderItemStatus
      Error: Pre-requisite step failed: event not found
      Error: Expected B request to be successful.
>> Book Only >> Book Only Success for random Slot >> B >> validation of B
  ❌️ passes validation checks
      Error: Pre-requisite step failed: event not found
      Error: No response to validate

======

Ran 22 tests in 5.367 s
❌ 22 failing
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
```

**Comment by ylt** [comment]
# OA-Test-BookCCancel-Paid-Failure

End to end booking and cancellation. Cancellation is expected to fail as Opportunity.startDate is in the past.

## Test setup
### create-session Request
POST https://bookingsystemreferenceimplementation.azurewebsites.net/api/openbooking/test-interface/ScheduledSession
```json
{
    "@context": "https://openactive.io/",
    "@type": "ScheduledSession",
    "superEvent": {
        "@type": "SessionSeries",
        "name": "OA-Test-BookCCancel-Paid-Failure",
        "offers": [
            {
                "@type": "Offer",
                "price": 14.95
            }
        ]
    },
    "startDate": "2020-03-05T11:23:35+00:00",
    "endDate": "2020-03-06T11:23:35+00:00",
    "maximumAttendeeCapacity": 5
}
```

---

Response status code: 200 OK. Responded in 519.828002ms
```json
{
    "@context": "https://openactive.io/",
    "@type": "ScheduledSession",
    "@id": "https://bookingsystemreferenceimplementation.azurewebsites.net/api/identifiers/scheduled-sessions/101138/events/101139"
}
```
### get-match Request
GET http://localhost:3000/get-cached-opportunity/https%3A%2F%2Fbookingsystemreferenceimplementation.azurewebsites.net%2Fapi%2Fidentifiers%2Fscheduled-sessions%2F101138%2Fevents%2F101139
**Response still pending**
### delete-session Request
DELETE https://bookingsystemreferenceimplementation.azurewebsites.net/api/openbooking/test-interface/ScheduledSession/https%3A%2F%2Fbookingsystemreferenceimplementation.azurewebsites.net%2Fapi%2Fidentifiers%2Fscheduled-sessions%2F101138%2Fevents%2F101139

---

Response status code: 404 Not Found. Responded in 464.02878ms
## C1
### Specs

File: packages/openactive-integration-tests/test/report-generator.js [hunk 1]
```diff
@@ -18,6 +19,12 @@ class ReportGenerator {
       return chalkFn(options.fn(this));
     });
 
+    Handlebars.registerHelper("renderSuiteName", function(suiteName, options) {
+      if (suiteName.length <= 2) return "Test setup";
+
+      return suiteName.slice(2).join(" >> ");
+    });
```
Comment (henryaddison) [inline packages/openactive-integration-tests/test/report-generator.js]: I can't remember the output from before. Are the first two parts of the suite name just not useful, common phrases?

File: packages/openactive-integration-tests/test/helpers/logger.js [hunk 2]
```diff
@@ -166,6 +166,20 @@ class ReporterLogger extends BaseLogger {
     return this.testName;
   }
 
+  get activeSuites() {
```
Comment (henryaddison) [inline packages/openactive-integration-tests/test/helpers/logger.js]: So is this to limit to suites that were actually run (hence sourcing them from logs) as opposed to all the tests suites registered in the spec files?

Comment (ylt) [reply packages/openactive-integration-tests/test/helpers/logger.js]: activeSuites - the idea is to only list suites with `it` blocks in, the suites array even contains suites that only contain describe() blocks (ended up with loads of section headers with no content)

File: packages/openactive-integration-tests/test/report-generator.js (hunk 1 above)
Comment (ylt) [reply packages/openactive-integration-tests/test/report-generator.js]: the first two parts of suite name are already referenced by the filename, and were just extra noise

**Pull request merged**